streaming/base/dataset.py

```diff
@@ -743,7 +742,8 @@ def _generate_work(self, world: World, epoch: int, sample_in_epoch: int) -> NDAr
         """
         # Ensure that num_canonical_nodes has been set.
         if self.num_canonical_nodes is None:
-            raise RuntimeError('Number of canonical nodes can never be None')
+            raise RuntimeError(f'`num_canonical_nodes` can never be None. ' +
+                               f'Provide a positive number.')
```

Comment:
number -> integer?

Reply:
Updated in https://github.com/mosaicml/streaming/pull/298/commits/ffb9eb9140df12f9f855fc02b82bf98c60ba776b.